File: src/main/java/xyz/upperlevel/hgame/input/EntityInput.java
package xyz.upperlevel.hgame.input;

import java.util.ArrayList;
import java.util.List;

import static java.util.Collections.unmodifiableList;

public class EntityInput {
    private final List<InputAction> actions;
    
    public EntityInput(List<InputAction> actions) {
        this.actions = new ArrayList<>(actions);
    }

    public List<InputAction> getActions() {
        return unmodifiableList(actions);
    }

    public void onNetworkAction(int actionType) {
        actions.stream()
                .filter(a -> a.getId() == actionType)
                .findAny()
                .orElseThrow(() -> new IllegalArgumentException("Illegal action: " + actionType))
                .onTrigger();
    }
}

File: src/main/java/xyz/upperlevel/hgame/input/EntityInput.kt
package xyz.upperlevel.hgame.input

import java.util.*
import java.util.Collections.unmodifiableList

class EntityInput(actions: List<InputAction>) {
    private val actions: List<InputAction>

    init {
        this.actions = ArrayList(actions)
    }

    fun getActions(): List<InputAction> {
        return unmodifiableList(actions)
    }

    fun onNetworkAction(actionType: Int) {
        actions.stream()
                .filter { a -> a.id == actionType }
                .findAny()
                .orElseThrow { IllegalArgumentException("Illegal action: $actionType") }
                .onTrigger()
    }
}
